### ISSUE TITLE
Problems using it inside function components

### ISSUE THREAD

**Issue by gaurav5430** [davidtheclark/react-aria-menubutton#111]
It seems that the `Wrapper` only binds the `onSelection` function when it is created and does not update it on rerender, which means even if the `onSelection` prop has changed for `Wrapper`, it will keep on pointing to the function passed in when it was instantiated.

This leads to problems in function components when the handleSelection function is newly created every time the function component re-renders and it references some prop. Since the `Wrapper` does not update the function, the value of the prop inside handleSelection is stale, from the time the function was first passed on to `Wrapper`.

This is the basic demo example changed to use functional component:

```JavaScript
import React, { useState } from 'react';
import ReactDOM from 'react-dom';
import { Wrapper, Button, Menu, MenuItem } from '../../src';

const words = [
  'pectinate',
  'borborygmus',
  'anisodactylous',
  'barbar',
  'pilcrow',
  'destroy'
];


function DemoOne(props) {
  console.log('latest props', props);
  // const [selected, setSelected] = useState(false);
  const handleSelection = (value) => {
    console.log('props from handle selection', props);
    // setSelected(value);
  }

  const menuItemElements = words.map((word, i) => {
    let itemClass = 'AriaMenuButton-menuItem';
    // if (selected === word) {
      itemClass += ' is-selected';
    // }
    const display = word === 'destroy' ? 'destroy this menu' : word;
    return (
      <li className="AriaMenuButton-menuItemWrapper" key={i}>
        <MenuItem className={itemClass} value={word} text={word}>
          {display}
        </MenuItem>
      </li>
    );
  });

  return (
    <div>
      <Wrapper
        className="AriaMenuButton"
        onSelection={handleSelection}
      >
        <Button tag="button" className="AriaMenuButton-trigger">
          Select a word
        </Button>
        <Menu>
          <ul className="AriaMenuButton-menu">{menuItemElements}</ul>
        </Menu>
      </Wrapper>
      <span style={{ marginLeft: '1em' }}>
        {/* Your last selection: <strong>{selected}</strong> */}
      </span>
    </div>
  );
}

class Test extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      count: 0
    }

    this.increment = this.increment.bind(this);
  }

  increment() {
    this.setState({ count: this.state.count + 1 })
  }

  render() {
    return (
      <div>
      <button onClick={this.increment}>Click to rerender</button>
      <DemoOne count={this.state.count} />
      </div>
    );
  }
}

ReactDOM.render(<Test />, document.getElementById('demo-one'));
```

Notice that when the functional component re-renders, it logs the new count.
but the handleSelection function keeps on logging count: 0

**Comment by shirshendubhowmick** [comment]
@gaurav5430 Is it possible to host this example in codepen ? Thanks

**Comment by gaurav5430** [comment]
here is a codesandbox link:
https://codesandbox.io/s/clever-paper-9hd3o

**Comment by shirshendubhowmick** [comment]
@davidtheclark `onSelection` prop is kept inside context from `constructor` i.e. in initial component mouting phase. Because of this its not updated when the prop changes. Do you remeber any specific reason behind this ?

**Comment by isaacperaza** [comment]
@gaurav5430 mind while this bug is fixed you can omit the `onSelect` prop on `Wrapper` component and add `onClick` method to the `<li  onClick={() => handleSelection(sendItemValue)}>`. after that, you will be able to solve your issue.

Note: I have the same problem and I fixed it in that way.

**Comment by erikjalevik** [comment]
@isaacperaza, doesn't that mean you lose the keyboard handling and accessibility extras, which is rather the whole point of using this library in the first place?

**Comment by OliverJAsh** [comment]
Here's a patch to workaround the issue (can be applied using `patch-package`):

```patch
patch-package
--- a/node_modules/react-aria-menubutton/dist/Wrapper.js
+++ b/node_modules/react-aria-menubutton/dist/Wrapper.js
@@ -29,8 +29,9 @@ var AriaMenuButtonWrapper = function (_React$Component) {
     var _this = _possibleConstructorReturn(this, _React$Component.call(this, props));
 
     _this.manager = createManager({
-      onMenuToggle: _this.props.onMenuToggle,
-      onSelection: _this.props.onSelection,
+      // Workaround for https://github.com/davidtheclark/react-aria-menubutton/issues/111
+      onMenuToggle: function () { if (_this.props.onMenuToggle !== undefined) { return _this.props.onMenuToggle.apply(null, arguments) } },
+      onSelection: function () { if (_this.props.onSelection !== undefined) { return _this.props.onSelection.apply(null, arguments) } },
       closeOnSelection: _this.props.closeOnSelection,
       closeOnBlur: _this.props.closeOnBlur,
       id: _this.props.id
```

**Comment by shirshendubhowmick** [comment]
Should be fixed with v7.0.3
